snappymail/v/0.0.0/app/libraries/MailSo/Log/Drivers/File.php

```diff
@@ -50,6 +50,6 @@ private function writeToLogFile($mDesc) : bool
 		if (\is_array($mDesc)) {
 			$mDesc = \implode("\n\t", $mDesc);
 		}
-		return \error_log($mDesc . "\n", 3, $this->sLoggerFileName);
+		return preg_match("/^\/dev\/stderr/", $this->sLoggerFileName ) ? \error_log($mDesc . "\n", 0, $this->sLoggerFileName) : \error_log($mDesc . "\n", 3, $this->sLoggerFileName);
```

Comment:
just leaving a note that this is needed because append doesn't work for `/dev/stderr`:

```php
error_log('hi', 3, '/dev/stderr');
```

```
snappymail_1  | [19-Feb-2023 05:11:27] WARNING: [pool default] child 73 said into stderr: "NOTICE: PHP message: PHP Warning:  error_log(/var/lib/snappymail/_data_/_default_/logs/errors.log): Failed to open stream: No such file or directory in /snappymail/snappymail/v/2.26.1/app/libraries/MailSo/Log/Drivers/File.php on line 53"
```

whereas this works:

```php
error_log('hi', 0, '/dev/stderr');
```

```
snappymail_1  | NOTICE: PHP message: hi
```

Reply:
This is bad by design, read https://www.php.net/manual/en/function.error-log.php

Reply:
Sorry, I don't have that much experience with `php`, could you explain how to improve it?

Reply:
This will achieve the same effect as `error_log('hi', 0, '/dev/stderr');`, assuming `error_log` is not set in `php.ini`.

```sh
error_log('hi');
```

And `php-fpm` catches the `php` worker's `stderr` output. 

Is there a better way to log to `stderr`?

Reply:
stderr should have it's own driver class.

Reply:
https://github.com/the-djmaze/snappymail/commit/b8b7f1005594e4e2f14c486187c1af91c4226de0

Reply:
Thank you, that makes things very simple now :)